## Задача 1

Даны значения величины заработной платы заемщиков банка (salary) и значения их поведенческого кредитного скоринга (scoring):

salary = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
scoring = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

1. Найдите ковариацию этих двух величин: сначала без использования специальных функций, а затем с помощью функции numpy.cov. Полученные значения должны быть равны.

In [64]:
import pandas as pd
from scipy import stats
import numpy as np
import warnings
warnings.filterwarnings("ignore")

salary = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110] 
scoring = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

# создадим датафрейм
borrower=pd.DataFrame(zip(salary, scoring), columns = ['salary', 'scoring'])

# найдем среднее
Msalary = borrower["salary"].mean()
Mscoring = borrower["scoring"].mean()

cov = ((borrower["salary"] - Msalary) * (borrower["scoring"] - Mscoring)).sum() / (borrower["salary"].count() - 1)
print("Несмещенная ковариация: ",cov)

Несмещенная ковариация:  10175.37777777778


Найдем и проверим с помощю встроенных функций numpy

In [28]:
np.cov(borrower["salary"],borrower["scoring"])

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

В матрице ковариаций надо выбирать значение вне диагонали (10175.37777778). Полностью совпадает с ручным расчетом.


2. Найдите коэффициент корреляции Пирсона: с помощью ковариации и средних квадратических отклонений двух признаков, а затем с помощью специальных функций из библиотек numpy и pandas.

Используем формулу: $$r_{XY} = \dfrac{\sigma_{XY}}{\sigma_X \cdot \sigma_Y}$$

In [30]:
corr = cov / (borrower["salary"].std() * borrower["scoring"].std(ddof = 1))
corr

0.8874900920739164

Найдем и проверим с помощю встроенных функций numpy

In [39]:
corrnump=np.corrcoef(borrower["salary"],borrower["scoring"])
corrnump

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

Значение совпадают. 

## Задача 2

Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111

Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [51]:
iq=pd.DataFrame([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
n = iq[0].count()
mean = iq[0].mean()
std = iq[0].std(ddof=1)

n, mean, std

(10, 118.1, 10.54566788359614)

In [54]:
p = 0.95
alpha = 1 - p

t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-2.2621571627409915, 2.2621571627409915)

In [57]:
print("Доверительный интервал: ",(mean + t1 * std / np.sqrt(n), mean + t2 * std / np.sqrt(n)))

Доверительный интервал:  (110.55608365158724, 125.64391634841274)


## Задача 3

Известно, что рост футболистов в сборной распределен нормально с известной дисперсией 25. На выборке объёма 27 выборочное среднее составило 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Т.к. выборка нормально распределённая, то используем формулу: 
$$Z = \dfrac{\overline{X} - \mu}{\sigma / \sqrt{n}}$$

In [62]:
n=27 # выборка
D = 25 # дисперсия
std = np.sqrt(D) # СКО
mean = 174.2 # среднее ариф. роста
p=0.95 # надежность p

alpha = 1 - p

t = stats.norm.ppf(1 - alpha/2)
t

1.959963984540054

In [63]:
print("Доверительный интервал: ",(mean - t * std / np.sqrt(n), mean + t * std / np.sqrt(n)))

Доверительный интервал:  (172.3140237765397, 176.08597622346028)
